SoftLayer/CLI/dedicatedhost/cancel.py

```diff
@@ -0,0 +1,33 @@
+"""Cancel a dedicated server."""
+# :license: MIT, see LICENSE for more details.
+
+import click
+
+import SoftLayer
+from SoftLayer.CLI import environment
+from SoftLayer.CLI import exceptions
+from SoftLayer.CLI import formatting
+from SoftLayer.CLI import helpers
+
+
+@click.command()
+@click.argument('identifier')
+@click.option('--immediate',
+              is_flag=True,
+              default=False,
+              help="Cancels the server immediately (instead of on the billing anniversary)")
+@click.option('--comment',
+              help="An optional comment to add to the cancellation ticket")
+@click.option('--reason',
+              help="An optional cancellation reason. See cancel-reasons for a list of available options")
+@environment.pass_env
+def cli(env, identifier, immediate, comment, reason):
+    """Cancel a dedicated server."""
+
+    mgr = SoftLayer.DedicatedHostManager(env.client)
+    host_id = helpers.resolve_id(mgr.resolve_ids, identifier, 'dedicatedhost')
```

Comment:
should be 
```
order_id = helpers.resolve_id(mgr.resolve_ids, host_id, 'dedicatedhost')
```

Reply:
Yeah. Pretty sure there are three ids.
1. slcli dedicatedhost create  => return order_id
2. slcli dedicatedhost list => return list of host_ids
3. slcli dedicatedhost cancel host_id => it will resolve host_id to `some_id` 

e.g.
1. 
``` slcli dedicatedhost create --hostname yingchao-test-host-3 --domain rescale.com --datacenter wdc07 --flavor 56_CORES_X_242_RAM_X_1_4_TB ```
return  `id : 29704621` which is order_id.

2.  ```slcli dedicatedhost list ``` return `244511 ` which is host_id

3. ```slcli dedicatedhost cancel 244511 --immediate``` it will resolve id to some_id, which i believe is `349983939`.

{"showBluemixLinking":false,"showLinkToBluemix":false,"success":true,"result":{"id":244511,"name":"yingchao-test-host-3","billingItem":{"categoryCode":"dedicated_virtual_hosts","hourlyRecurringFee":"3.164","id":349983939,"activeAssociatedChildren":[{"description":"242 GB RAM","id":349983941,"parentId":349983939,"category":{"name":"Dedicated Host RAM"}},{"description":"1.2 TB Local Storage (Dedicated Host Capacity)","id":349983943,"parentId":349983939,"category":{"name":"Dedicated Host Disk"}}],"activeChildren":[{"description":"242 GB RAM","id":349983941,"parentId":349983939,"category":{"name":"Dedicated Host RAM"}},{"description":"1.2 TB Local Storage (Dedicated Host Capacity)","id":349983943,"parentId":349983939,"category":{"name":"Dedicated Host Disk"}}],"hourlyFlag":true},"datacenter":{"longName":"Washington 7"},"associatedItems":[],"fullyQualifiedDomainName":"yingchao-test-host-3","anniversaryDate":"2018-09-30T23:00:00-06:00","cutoffDate":"2018-09-30T22:59:59-06:00","allowImmediateCancellation":true}}

Reply:
Anyway, it is not that important. Just want to avoid naming confusion.